### PR TITLE
adding support for SciTokens https://scitokens.org/

### PR DESCRIPTION
**Pull request by ag012**

### CODE REVIEW

File: server/packaging/fedora/oauth-ssh.spec.in [hunk 1]
```diff
@@ -30,6 +30,7 @@ BuildRequires:  libtool-ltdl-devel
 BuildRequires:  checkpolicy
 BuildRequires:  policycoreutils-python
 BuildRequires:  python-setuptools
+BuildRequires:  scitokens-cpp
```
Comment (jbasney) [inline server/packaging/fedora/oauth-ssh.spec.in]: The scitokens-cpp package can be built from source (https://github.com/scitokens/scitokens-cpp) or RPMs are available from OSG (https://opensciencegrid.org/docs/common/yum/) and soon from EPEL (https://src.fedoraproject.org/rpms/scitokens-cpp).

File: server/README.md [hunk 2]
```diff
@@ -310,6 +310,17 @@ pam_oauth_ssh.so returns the following control values:
...
+## SciTokens
+
+To enable SciToken as a verification method:
+1. Compile the module from the source.
+2. Set allowed issuers in oauth-ssh.conf file
+3. Install client-side normally and use SciToken to login. (Both valid SciToken and Globus Auth token will be accepted)
+
+Payload format for Scitokens:
+  "scope": “ssh:username",
+  "aud": "demo.scitokens.org"
```
Comment (jbasney) [inline server/README.md]: I think the audience (aud) is the hostname of the SSH server, based on the gethostname() call in scitoken_verify().

**Comment by ag012** [comment]
> What is auth_method? Is it already split by space?
auth_method is a member of config. config is a struct preprocessed in config.c


> Probably just need this to be "break", since it will break out of the loop right to the goto statement.
Thanks for the comment. I've change that to break.

**Comment by ag012** [comment]
Note that rewrite _cmd_login 78f9c13 added a non-optional configuration field and it should be mentioned outside of the scitokens section in readme

**Comment by rpwagner** [comment]
> Thanks @ag012! I submitted a corresponding XSEDE activity for this feature at https://software.xsede.org/display/xci-694.

@jbasney this may be both overly pedantic and not be the right place to discuss this, but on the XSEDE activity the term "proprietary" seems inappropriate for Globus Auth OAuth 2.0 Tokens. They're opaque bearer tokens based on [IETF RFC 7662](https://tools.ietf.org/html/rfc7662). Opaque is not the same as proprietary.

**Comment by jbasney** [comment]
> @jbasney this may be both overly pedantic and not be the right place to discuss this, but on the XSEDE activity the term "proprietary" seems inappropriate for Globus Auth OAuth 2.0 Tokens. They're opaque bearer tokens based on [IETF RFC 7662](https://tools.ietf.org/html/rfc7662). Opaque is not the same as proprietary.

I'm happy to be corrected. It's my understanding that only tokens issued by globus.org are currently accepted by oauth-ssh, but I agree it should be possible to support any issuers of opaque bearer tokens that comply with [RFC 7662](https://tools.ietf.org/html/rfc7662) by adding a few new oauth-ssh configuration options. I opened issue #70 for it. In that case, I'd agree the term "proprietary" would no longer be appropriate.

**Comment by rpwagner** [comment]
Thanks, Jim. I'll contribute to #70.

**Comment by JasonAlt** [comment]
If this is ready approved and ready for merge, it'll need the 'draft' flag changed.

**Comment by jbasney** [comment]
Who can remove the draft flag? Here's what I see:

image

**Pull request merged**